modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java

```diff
@@ -16,167 +17,204 @@
-     * Gets the document number from database.
-     *
-     * @param conn
-     *          Handler for the database connection.
-     * @param AD_Client_ID
-     *          String with the client id.
-     * @param TableName
-     *          Table name.
-     * @param updateNext
-     *          Save the new sequence in database.
-     * @return String with the new document number.
-     */
-    @Override
-    public String getDocumentNo(ConnectionProvider conn, String AD_Client_ID, String TableName,
-                                       boolean updateNext) {
-        if (TableName == null || TableName.length() == 0) {
-            throw new IllegalArgumentException("Utility.getDocumentNo - required parameter missing");
-        }
-
-        CSResponse cs = null;
-        try {
-            cs = DocumentNoData.nextDoc(conn, "DocumentNo_" + TableName, AD_Client_ID,
-                    (updateNext ? "Y" : "N"));
-        } catch (final ServletException e) {
-        }
-
-        if (cs == null || cs.razon == null) {
-            return "";
-        } else {
-            return cs.razon;
-        }
-    }
-
-    /**
-     * Gets the document number from database.
-     *
-     * @param conn
-     *          Handler for the database connection.
-     * @param AD_Client_ID
-     *          String with the client id.
-     * @param TableName
-     *          Table name.
-     * @param updateNext
-     *          Save the new sequence in database.
-     * @return String with the new document number.
-     */
-    @Override
-    public String getDocumentNoConnection(Connection conn, ConnectionProvider con,
-                                                 String AD_Client_ID, String TableName, boolean updateNext) {
-        if (TableName == null || TableName.length() == 0) {
-            throw new IllegalArgumentException("Utility.getDocumentNo - required parameter missing");
-        }
-
-        CSResponse cs = null;
-        try {
-            cs = DocumentNoData.nextDocConnection(conn, con, "DocumentNo_" + TableName, AD_Client_ID,
-                    (updateNext ? "Y" : "N"));
-        } catch (final ServletException e) {
-        }
-
-        if (cs == null || cs.razon == null) {
-            return "";
-        } else {
-            return cs.razon;
-        }
+  /**
+   * Gets the document number from the database.
+   *
+   * @param conn
+   *     Handler for the database connection.
+   * @param vars
+   *     Handler for the session info.
+   * @param WindowNo
+   *     Window id.
+   * @param TableName
+   *     Table name.
+   * @param C_DocTypeTarget_ID
+   *     Id of the doctype target.
+   * @param C_DocType_ID
+   *     id of the doctype.
+   * @param onlyDocType
+   *     Search only for doctype.
+   * @param updateNext
+   *     Save the new sequence in database.
+   * @return String with the new document number.
+   */
+  @Override
+  public String getDocumentNo(ConnectionProvider conn, VariablesSecureApp vars, String WindowNo,
```

Comment:
vars names are in mixed case with a lowercase first letter.

https://www.oracle.com/java/technologies/javase/codeconventions-namingconventions.html